chia/wallet/puzzles/pool_member_innerpuz.clvm

```diff
@@ -0,0 +1,29 @@
+(mod (POOL_PUZHASH RELATIVE_LOCK_HEIGHT ESCAPE_MODE_PUZHASH P2_SINGLETON_PUZHASH PUBKEY spend_type my_puzhash my_amount pool_reward_amount pool_reward_height)
+
+  ; POOL_PUZHASH is commitment to the pool's puzzle hash
+  ; RELATIVE_LOCK_HEIGHT is how long it takes to leave
+  ; ESCAPE_MODE_PUZHASH is the puzzle_hash you'll go to when you iniate the leaving process
+  ; P2_SINGLETON_PUZHASH is the puzzlehash for your pay to singleton puzzle
+
+  ; spend_type is: 0 for absorbing money, 1 to escape
+
+  ; absorb_coin_id and my_puzhash are only used for absorbtion case
+
+  (include condition_codes.clvm)
+  (defconstant GENESIS_CHALLENGE 0xccd5bb71183532bff220ba46c268991a)
+
+  (defun calculate_pool_reward (pool_reward_height P2_SINGLETON_PUZHASH pool_reward_amount)
+    (sha256 (concat (lsh GENESIS_CHALLENGE (- 128 (* (strlen pool_reward_height) 8))) pool_reward_height) P2_SINGLETON_PUZHASH pool_reward_amount)
```

Comment:
You might consider rewriting `(lsh GENESIS_CHALLENGE (- 128 (* (strlen pool_reward_height) 8))) pool_reward_height)` as an inline function with a name that explains what's going on. Example:

```
(defun-inline pool-reward-parent-puzzle-hash (height)
    (concat (lsh GENESIS_CHALLENGE (- 128 (* (strlen height) 8))) height)))
```

But this uses `concat`, and so is very suspicious.

I suspect something like this would work better and be a lot simpler:

```
(defun-inline pool-reward-parent-puzzle-hash (height)
    (logior (ash GENESIS_CHALLENGE 128) height)
```